mapie/metrics.py

```diff
@@ -307,3 +313,426 @@ def top_label_ece(
+
+    Warning: This metric should be used only with non constant intervals
+    (intervals of different sizes), with constant intervals the result
+    may be misinterpreted.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_intervals: NDArray of shape (n_samples, 2, n_alpha) or (n_samples, 2)
+        Prediction intervals given by booleans of labels.
+    num_bins: int n
+        Number of groups. Should be less than the number of different
+        interval widths.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha,)
+
+    Examples
+    --------
+    >>> from mapie.metrics import regression_ssc
+    >>> import numpy as np
+    >>> y_true = np.array([5, 7.5, 9.5])
+    >>> y_intervals = np.array([
+    ... [[4, 4], [6, 7.5]],
+    ... [[6.0, 8], [9.0, 10]],
+    ... [[9, 9], [10.0, 10.0]]
+    ... ])
+    >>> print(regression_ssc_score(y_true, y_intervals, num_bins=2))
+    [1.  0.5]
+    """
+    return np.min(regression_ssc(y_true, y_intervals, num_bins), axis=1)
+
+
+def classification_ssc(
+    y_true: NDArray,
+    y_pred_set: NDArray,
+    num_bins: Union[int, None] = None
+) -> NDArray:
+    """
+    Compute Size-Stratified Coverage metrics proposed in [3] that is
+    the conditional coverage conditioned by the size of the predictions sets.
+    The sets are ranked by their size (ascending) and then divided into
+    num_bins groups: one value of coverage by groups is computed.
+
+    [3] Angelopoulos, A. N., & Bates, S. (2021).
+    A gentle introduction to conformal prediction and
+    distribution-free uncertainty quantification.
+    arXiv preprint arXiv:2107.07511.
+
+    Parameters
+    ----------
+    y_true: NDArray of shape (n_samples,)
+        True labels.
+    y_pred_set: NDArray of shape (n_samples, n_class, n_alpha)
+    or (n_samples, n_class)
+        Prediction sets given by booleans of labels.
+    num_bins: int or None
+        Number of groups. If None, one value of coverage by possible
+        size of sets (n_classes +1) is computed. Should be less than the
+        number of different set sizes.
+
+    Returns
+    -------
+    NDArray of shape (n_alpha, num_bins)
+
+    Examples
+    --------
+    >>> from mapie.metrics import classification_ssc
+    >>> import numpy as np
+    >>> y_true = y_true_class = np.array([3, 3, 1, 2, 2])
+    >>> y_pred_set = np.array([
+    ...    [True, True, True, True],
+    ...    [False, True, False, True],
+    ...    [True, True, True, False],
+    ...    [False, False, True, True],
+    ...    [True, True, False, True]])
+    >>> print(classification_ssc(y_true, y_pred_set, num_bins=2))
+    [[1.         0.66666667]]
+    """
+    y_true = cast(NDArray, column_or_1d(y_true))
+    y_pred_set = check_array_shape_classification(y_true, y_pred_set)
+
+    sizes = np.sum(y_pred_set, axis=1)
+    n_classes = y_pred_set.shape[1]
+    if num_bins is None:
+        bins = list(range(n_classes + 1))
```

Comment:
By doing so, in the case where you may have empty bins (if for instance their is no prediction sets with 5 labels). Is it possible to create the bins according the actual sizes that exists ?

Reply:
As it is actually implemented it returns nan if there is an empty bin. It is possible to change it and create the bins according the actual sizes that exists but the user will not know which bin is empty (which prediction set size does not exists) (it is not necessarily a problem)